dev-tools/mage/manifest/manifestspecs.go

```diff
@@ -0,0 +1,163 @@
+package manifest
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/magefile/mage/mg"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"time"
+)
+
+type Dependency struct {
+	Prefix   string `json:"prefix"`
+	BuildUri string `json:"build_uri"`
+}
+
+type Project struct {
+	Branch                       string             `json:"branch"`
+	CommitHash                   string             `json:"commit_hash"`
+	CommitUrl                    string             `json:"commit_url"`
+	ExternalArtifactsManifestUrl string             `json:"external_artifacts_manifest_url"`
+	BuildDurationSeconds         int                `json:"build_duration_seconds"`
+	Packages                     map[string]Package `json:"packages"`
+	Dependencies                 []Dependency       `json:"dependencies"`
+}
+
+type Package struct {
+	Url          string   `json:"url"`
+	ShaUrl       string   `json:"sha_url"`
+	AscUrl       string   `json:"asc_url"`
+	Type         string   `json:"type"`
+	Architecture string   `json:"architecture"`
+	Os           []string `json:"os,omitempty"`
+	Classifier   string   `json:"classifier,omitempty"`
+	Attributes   struct {
+		IncludeInRepo string `json:"include_in_repo,omitempty"`
+		Oss           string `json:"oss,omitempty"`
+		ArtifactNoKpi string `json:"artifactNoKpi,omitempty"`
+		Internal      string `json:"internal,omitempty"`
+		ArtifactId    string `json:"artifact_id,omitempty"`
+		Group         string `json:"group,omitempty"`
+	} `json:"attributes,omitempty"`
+}
+
+type ManifestResponse struct {
+	Branch               string             `json:"branch"`
+	ReleaseBranch        string             `json:"release_branch"`
+	Version              string             `json:"version"`
+	BuildId              string             `json:"build_id"`
+	StartTime            string             `json:"start_time"`
+	EndTime              string             `json:"end_time"`
```

Comment:
I would have preferred for time to be represented as `time.Time` but depending on which format we use here we may need a custom json marshaler/unmarshaler

Reply:
going to remove this in favour of the artifacts_api struct